scripts/assets/test_file_exemptions.textproto

```diff
@@ -807,3 +808,11 @@ exempted_file_path: "utility/src/main/java/org/oppia/android/util/threading/Conc
 exempted_file_path: "utility/src/main/java/org/oppia/android/util/threading/DispatcherInjector.kt"
 exempted_file_path: "utility/src/main/java/org/oppia/android/util/threading/DispatcherInjectorProvider.kt"
 exempted_file_path: "utility/src/main/java/org/oppia/android/util/threading/DispatcherModule.kt"
+exempted_file_path: "utility/src/main/java/org/oppia/android/util/logging/performancemetrics/PerformanceMetricsUtilsModule.kt"
+exempted_file_path: "utility/src/main/java/org/oppia/android/util/logging/performancemetrics/PerformanceMetricsUtils.kt"
+exempted_file_path: "utility/src/main/java/org/oppia/android/util/logging/MetricLogScheduler.kt"
+exempted_file_path: "utility/src/main/java/org/oppia/android/util/logging/performancemetrics/MetricLogSchedulerModule.kt"
+exempted_file_path: "utility/src/main/java/org/oppia/android/util/logging/performancemetrics/PerformanceMetricsEventLogger.kt"
+exempted_file_path: "utility/src/main/java/org/oppia/android/util/logging/performancemetrics/PerformanceMetricsLogScheduler.kt"
+exempted_file_path: "domain/src/main/java/org/oppia/android/domain/oppialogger/logscheduler/MetricLogSchedulingWorkerFactory.kt"
+exempted_file_path: "domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/PerformanceMetricsLoggerModule.kt"
```

Comment:
Here & elsewhere: it seems like all of these could have tests. Why can't we add test suites for each?

Reply:
`PerformanceMetricsEventLogger`, `PerformanceMetricsLogScheduler`, `PerformanceMetricsLoggerModule` are interfaces and `MetricLogSchedulingWorkerFactory` is marked exempted following the precedence of `LogUploadWorkerFactory`.

Reply:
Unless I'm missing something, ``PerformanceMetricsLogScheduler`` is not an interface. Also, ``PerformanceMetricsLoggerModule`` is a Dagger module so it's definitely testable (i.e. to ensure that the bindings that it's setting up a correct)--it's not an interface in the traditional sense since there's code gen involved here.

Precedence in this case actually isn't a good practice to follow. Much of these exemptions are not desired, many of them are "grandfathered" in so that the check could be introduced in CI (since otherwise introducing several hundred tests would have pushed back our ability to benefit from the check earlier). Instead, we should only continue to exempt classes that truly have no possible tests (such as pure interfaces) or cannot be tested effectively (the worker factory *may* fall under this, but I'm not certain).

Reply:
Added tests for `PerformanceMetricsLoggerModule`,  `MetricLogSchedulerModule` and `PerformanceMetricsLogScheduler`.